### PR TITLE
Fix SILGen of access to isolated static properties

### PR DESCRIPTION
**Pull request by kavon**
When accessing a static property isolated to a
global-actor, such as MainActor, a `hop_to_executor`
was not being emitted. This was caught by an assertion
I had left to catch such cases, but of course in release
builds this will lead to incorrect SIL, etc.

- [x] Fix basic access of a simple stored property
- [x] Fix access of an optional reference type with chained lookup.
- [x] Prevent emission of double-hop on standard access of static global (the `getCount` case).

Resolves rdar://78292384

### CODE REVIEW

**Comment by kavon** [comment]
@swift-ci please smoke test

**Comment by kavon** [comment]
@swift-ci please smoke test

File: lib/SILGen/SILGenLValue.cpp [hunk 1]
```diff
@@ -576,7 +587,8 @@ static SILValue enterAccessScope(SILGenFunction &SGF, SILLocation loc,
   // Push a writeback to end it.
   auto accessedMV = ManagedValue::forLValue(addr);
   std::unique_ptr<LogicalPathComponent>
-    component(new EndAccessPseudoComponent(typeData));
+    component(new EndAccessPseudoComponent(typeData,
+                                              std::move(prevExecutor)));
```
Comment (ktoso) [inline lib/SILGen/SILGenLValue.cpp:591]: Nice, that's nicely modeled with the move and breadcrumb 👍

**Review by ktoso — APPROVED** [review]
Looks good to me 👍 Nicely modeled by the way.

**Comment by kavon** [comment]
@swift-ci please smoke test and merge

**Comment by kavon** [comment]
@swift-ci please smoke test and merge

**Pull request merged**